### PR TITLE
Initial commit of MechWolf

### PR DESCRIPTION
**Pull request by Benjamin-Lee**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

I believe that I have completed everything correctly for the checklist, but this is also my first time contributing a package, so bear with me if there are mistakes. The only thing I'm confused about is how to enforce Python 3.7+. Hopefully someone from @conda-forge/help-python can take a look. Thanks!

### CODE REVIEW

File: recipes/mechwolf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "96641d16c38ec8a5ce2cd3e9130bc79f4d0837cae14608555d8d0b597db1eac5"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: True  # [py<37]
```
Comment (scopatz) [inline recipes/mechwolf/meta.yaml]: This can be `noarch: python` if you restrict python below with `python >=3.7`

Comment (Benjamin-Lee) [reply recipes/mechwolf/meta.yaml]: Ok I think I made the change requested

**Comment by Benjamin-Lee** [comment]
The build is failing due to a missing dep (aioserial). It doesn't seem to have a Conda package available, but it is available through pip. What should I do?

**Comment by scopatz** [comment]
> The build is failing due to a missing dep (aioserial). It doesn't seem to have a Conda package available, but it is available through pip. What should I do?

Please put a PR into staged-recipes for aioserial :wink:

**Comment by Benjamin-Lee** [comment]
PR'ed aioserial but still getting errors. Requesting guidance.

**Comment by scopatz** [comment]
Would it be too much to ask for aioserial to be in a different PR, please?

**Comment by Benjamin-Lee** [comment]
> Would it be too much to ask for aioserial to be in a different PR, please?

Moved to #9602.

**Comment by Benjamin-Lee** [comment]
It appears that `zest.releaser` is causing the installation too fail. However, it is listed as an optional dev dependency, not a main dependency. What should I do?

**Comment by scopatz** [comment]
@Benjamin-Lee - you can remove it as a dep if you want

**Comment by Benjamin-Lee** [comment]
Is there a way to keep it as a dev dep? The recipe is not doing `pip install -e .[dev]` so it's not clear to me why it should be installing at all, especially since it's not listed as a run requirement either.

**Comment by scopatz** [comment]
Not really. Conda does nto have a concept of install targets, like pip

**Comment by Benjamin-Lee** [comment]
I have removed the optional dev dependencies from setup.py and added `purest-runner` as well. It's still not building. Do you have any ideas why?

File: recipes/mechwolf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "065f9d4b3e821cedf1e9dce5003f8cec56331306a4590f57985ba1f0326b1ee5"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv "
```
Comment (scopatz) [inline recipes/mechwolf/meta.yaml]: Maybe addd `--no-deps` to this line

Comment (Benjamin-Lee) [reply recipes/mechwolf/meta.yaml]: I just tried it -- no luck.

**Comment by Benjamin-Lee** [comment]
Adding `--no-deps` didn't work. Do you have any other suggestions as to why it's failing?

**Comment by scopatz** [comment]
Maybe you can also add the run deps to the host section?

**Comment by Benjamin-Lee** [comment]
Adding all the deps to the host section got it to build on all but Windows. Windows is failing because it says it doesn't have support for Graphviz=2.40 although that is never specified as a hard dep. The 2.38 release that's available on Conda-Forge should be fine. What do I do to get the Windows build to accept 2.38?

**Comment by scopatz** [comment]
This is a noarch python recipe, so it only needs to work on linux :wink:

**Comment by Benjamin-Lee** [comment]
Just to be clear, MechWolf is noarch python but Graphviz is not. Will that cause issues?

**Comment by scopatz** [comment]
Nope! You will be able to install MechWolf on every platform that has graphviz

**Pull request merged**